setup.py

```diff
@@ -197,13 +197,20 @@ def run(self):
     'testpath', 
 ]
 
-extras_require = setuptools_args['extras_require'] = {
+extra_requirements = {
     # FIXME: tests still require nose for some utility calls,
     # but we are running with pytest
-    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client'],
+    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client>=4.2'],
     'serve': ['tornado>=4.0'],
     'execute': ['jupyter_client>=4.2'],
 }
+s = [] 
+[s.extend(values) for values in extra_requirements.values()]
```

Comment:
I did not know that sum could be applied to lists. I'm intrigued because that could allow: `extra_requirements['all'] = sum(extra_requirements.values(), [])`. Which is perhaps a bit too clever but totally does the trick.